src/Headers.php

```diff
@@ -126,7 +125,7 @@ public static function fromString($string)
      *
      * @return $this
      */
-    public function setPluginClassLoader(PluginClassLocator $pluginClassLoader)
+    public function setPluginClassLoader(HeaderLoader $pluginClassLoader)
```

Comment:
Could we drop the setter here and instead add a constructor such as:

```
public function __construct() {
    $this->pluginClassLoader = new HeaderLoader();
}
```

That way, we can always guarantee the property is in a valid state and refer only to the property internally.

The getter (`getPluginClassLoader()`) could remain so that users can fetch the instance and mutate the hash map of header classes.

It may be preferable to add the loader as a required constructor argument so that it can be swapped out, but it looks like we only have one concrete implementation so for the time being this is probably unnecessary.

Reply:
can't it be an optional argument?

Reply:
Yes

Reply:
did a quick search on `setPluginClassLoader` in `./src` and `./test` and didn't find a match. There is one match in the `./docs`

Reply:
Yes, will leave as is for now